### PR TITLE
Feature/create sentence plan

### PR DESCRIPTION
**Pull request by SWhitlam**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java [hunk 1]
```diff
@@ -62,73 +64,123 @@ public Boolean getTspEligible() {
...
+                }
+            });
+
         }
         return assessmentNeeds;
     }
 
+
+    private boolean sectionIsOverThreshold(Section section) {
+        return Optional.ofNullable(section.getSectionOtherRawScore()).orElse(0L) >= Optional.ofNullable(section.getRefSection()).map(s->s.getRefCrimNeedScoreThreshold()).orElse(Long.MAX_VALUE);
```
Comment (llmikeyj) [inline src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java]: Is it safe to default this?

File: src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java [hunk 2]
```diff
@@ -62,73 +64,123 @@ public Boolean getTspEligible() {
...
+    }
+
+    private boolean sectionHasRiskOfReoffending(Section section, Map<String, String> planSection) {
+        return questionHasAnswer(Optional.ofNullable(section.getQuestions().get(Optional.ofNullable(planSection.get("reoffendingQuestion")).orElse(""))),"YES");
```
Comment (llmikeyj) [inline src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java]: and this, etc?

File: src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java (hunk 1 above)
Comment (SWhitlam) [reply src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java]: I think so, the scores are always positive numbers and the threshold will always be less than long.MAX.

File: src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java (hunk 2 above)
Comment (SWhitlam) [reply src/main/java/uk/gov/justice/digital/oasys/api/Assessment.java]: The answers in OASys are always YES or NO for these sections so safe to default to empty string when no answer is present as that will then set the risk flag to false by default. 

The key thing here is that for the assessment to be complete there will always be an answer except where there is bad data in OASys, the defaults are just guarding against that scenario. The alternative is to error and not return an assessment but an error response.

**Pull request merged**